### ISSUE TITLE
Possible precedence issue with control flow operator

### ISSUE THREAD

**Issue by eridal** [aspiers/stow#25]
Hi all,

Just recently installed a new system, and any stow command will output the same warning:

```
$ stow -V
Possible precedence issue with control flow operator at /usr/share/perl5/Stow.pm line 1736.
stow (GNU Stow) version 2.2.0
```

Is this known? Are there any workaround?

**Comment by kaushalmodi** [comment]
@eridal What's your perl version?

Here are my stow and perl versions:

```
> stow -V && perl --version
stow (GNU Stow) version 2.2.2

This is perl 5, version 16, subversion 3 (v5.16.3) built for x86_64-linux
```

**Comment by eridal** [comment]
@kaushalmodi here's the perl version

```
> stow -V && perl --version
stow (GNU Stow) version 2.2.0

This is perl 5, version 20, subversion 2 (v5.20.2) built for arm-linux-gnueabihf-thread-multi-64int
```

**Comment by aspiers** [comment]
Really sorry for the terribly slow response.  This was fixed in d788ce0c1c59b3158270143659f7a4363da73056 which was part of the 2.2.1 release.

**Comment by eridal** [comment]
Thanks! Confirming the problem is gone at v2.2.2

**Comment by kescobo** [comment]
This problem seems to have crept up again...

```
❯ stow --version && perl --version
Possible precedence issue with control flow operator (exit) at /usr/bin/stow line 839, <DATA> line 23.
stow (GNU Stow) version 2.4.0

This is perl 5, version 40, subversion 0 (v5.40.0) built for x86_64-linux-thread-multi
```

**Comment by aspiers** [comment]
That was already fixed by #113.

**Comment by F4LC0N69** [comment]
I have the same issue. Have I done something wrong??
I have folder structure as follows:-
`~/dotfiles/alacritty/.config/alacritty.toml`
And when I do following command in dotfiles directory
`stow alacritty`
`Possible precedence issue with control flow operator (exit) at /usr/bin/stow line 839, <DATA> line 23.`

`stow --version && perl --version`
gives
`Possible precedence issue with control flow operator (exit) at /usr/bin/stow line 839, <DATA> line 23.
stow (GNU Stow) version 2.4.0

This is perl 5, version 40, subversion 0 (v5.40.0) built for x86_64-linux-thread-multi

Copyright 1987-2024, Larry Wall

Perl may be copied only under the terms of either the Artistic License or the
GNU General Public License, which may be found in the Perl 5 source kit.

Complete documentation for Perl, including FAQ lists, should be found on
this system using "man perl" or "perldoc perl".  If you have access to the
Internet, point your browser at https://www.perl.org/, the Perl Home Page.`

Am I doing something wrong or is it a bug??

PS:-I'm on Arch, BTW

**Comment by aspiers** [comment]
It's not released yet, so unless you're installing from git, it's expected that you would still see the warning.

**Comment by aspiers** [comment]
It will be fixed in the 2.4.1 release but for now you can install from git or just manually patch your `stow` script with [this change](https://github.com/aspiers/stow/pull/113/files).

**Comment by noelmiller** [comment]
> It will be fixed in the 2.4.1 release but for now you can install from git or just manually patch your `stow` script with [this change](https://github.com/aspiers/stow/pull/113/files).

Is there any timeline on a fix for this? Fedora 41 is currently using the 2.4.0 package and I see that brew is using the 2.4.1 package.

**Comment by noelmiller** [comment]
> It's already fixed in 2.4.1 released in September.

Ok, so it sounds like I need to speak with the packager in Fedora to get it updated. Thanks so much!

**Comment by SebekGie** [comment]
if its already fixed so why i still got that massage in teminal?
